### PR TITLE
ci: migrate to circleci

### PR DESCRIPTION
**Pull request by atlawrie**
Migrates our CI pipeline from Travis to CircleCI.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-  # - 3.8
-  - pypy3
-
-env:
-  - TOX_ENV=core
-
-jobs:
-  include:
-    - python: 3.8
-      env: TOX_ENV=flake8
```
Comment (sherwinski) [inline .travis.yml:18]: I think we'll want to have at least one test [run the linter](https://github.com/imgix/imgix-python/blob/4f188b854f50a6cb942ef537588efcd79dfec6c2/CONTRIBUTING.md?plain=1) (flake8) just to match what CI was previously doing

**Review by sherwinski — CHANGES_REQUESTED** [review]

**Review by sherwinski — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+version: 2.1
+orbs:
+  python: circleci/python@2.1.1
+
+jobs:
+  test:
+    parameters:
+      version:
+        default: "current"
```
Comment (luqven) [inline .circleci/config.yml:9]: does "current" ever get used here? Should we default to the python version we're treating as LTS instead?

**Review by luqven — APPROVED** [review]
LGTM just one quick question.

File: .circleci/config.yml (hunk 2 above)
Comment (atlawrie) [reply .circleci/config.yml:9]: Oh that's a good catch. It should never actually fallback to the default, but you're absolutely right in that `current` won't be correctly interpreted as a version.

File: .travis.yml (hunk 1 above)
Comment (sherwinski) [reply .travis.yml:18]: Resolved offline, looks like it's already running in each job

**Pull request merged**